### PR TITLE
Updates Zephyr STM32F4 Discovery sample for Zephyr 1.13.99.

### PR DESCRIPTION
**Pull request by jwoolston**
As a first step towards #2449, I have updated the Zephyr STM32F4 Discovery sample so that it will compile with the latest tools: Kotlin Native 1.3.11 and Zephyr 1.13.99. This will still require setting the same environment variables as before.

### CODE REVIEW

File: samples/zephyr/c_interop/platforms/stm32f4_disco.bat [hunk 1]
```diff
@@ -17,21 +17,20 @@ if "%BOARD%" == "" (
...
-        -copt -I%ZEPHYR_BASE%/include/drivers ^
-        -copt -I%ZEPHYR_BASE%/ext/hal/cmsis/Include ^
-        -copt -I%ZEPHYR_BASE%/ext/hal/st/stm32cube/stm32f4xx/soc ^
-        -copt -I%ZEPHYR_BASE%/ext/hal/st/stm32cube/stm32f4xx/drivers/include ^
-        -copt -I%ZEPHYR_BASE%/ext/hal/st/stm32cube/stm32f4xx/drivers/include/Legacy ^
-        -copt -I%ZEPHYR_BASE%/drivers ^
-        -copt -I%DIR%/build/zephyr/include/generated ^
-        -copt -I%DIR%/build/zephyr/include/generated/syscalls ^
+        -compilerOpts '-Xclang -DSTM32F407xx' ^
+        -compilerOpts -I$ZEPHYR_BASE/kernel/include ^
```
Comment (msink) [inline samples/zephyr/c_interop/platforms/stm32f4_disco.bat]: %ZEPHYR_BASE%
and %BOARD%, %DIR%

File: samples/zephyr/c_interop/platforms/stm32f4_disco.bat [hunk 2]
```diff
@@ -17,21 +17,20 @@ if "%BOARD%" == "" (
...
-        -copt -I%ZEPHYR_BASE%/include ^
-        -copt -I%ZEPHYR_BASE%/include/drivers ^
-        -copt -I%ZEPHYR_BASE%/ext/hal/cmsis/Include ^
-        -copt -I%ZEPHYR_BASE%/ext/hal/st/stm32cube/stm32f4xx/soc ^
-        -copt -I%ZEPHYR_BASE%/ext/hal/st/stm32cube/stm32f4xx/drivers/include ^
-        -copt -I%ZEPHYR_BASE%/ext/hal/st/stm32cube/stm32f4xx/drivers/include/Legacy ^
-        -copt -I%ZEPHYR_BASE%/drivers ^
-        -copt -I%DIR%/build/zephyr/include/generated ^
-        -copt -I%DIR%/build/zephyr/include/generated/syscalls ^
+        -compilerOpts '-Xclang -DSTM32F407xx' ^
```
Comment (msink) [inline samples/zephyr/c_interop/platforms/stm32f4_disco.bat]: double quotes: "-Xclang -DSTM32F407xx"

**Comment by msink** [comment]
```

**Comment by msink** [comment]
After that changes - build passed, with a lot of warnings about `orphan section ....`

Kotlin:
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      224848 B         1 MB     21.44%
            SRAM:       17324 B       192 KB      8.81%
        IDT_LIST:         200 B         2 KB      9.77%
```
C samples/basic/blinky:
```
Memory region         Used Size  Region Size  %age Used
           FLASH:       13164 B         1 MB      1.26%
            SRAM:        4308 B       192 KB      2.19%
        IDT_LIST:         200 B         2 KB      9.77%
```

Cannot check if it works - do not have board currently.

File: samples/zephyr/src/main.kt [hunk 3]
```diff
@@ -8,16 +8,16 @@ import kotlinx.cinterop.*
...
 fun blinky(value: Int) {
 
-    val port = LED0_GPIO_PORT
+    val port = LED0_GPIO_CONTROLLER
     val led = LED0_GPIO_PIN
     var toggler = false
     val dev = device_get_binding(port)
 
-    gpio_pin_configure(dev, led, GPIO_DIR_OUT)
+    gpio_pin_configure(dev, led.toUInt(), GPIO_DIR_OUT)
```
Comment (olonho) [inline samples/zephyr/src/main.kt]: `convert()` is more generic primitive.

**Comment by olonho** [comment]
Please add smth like
```diff --git a/samples/zephyr/build.sh b/samples/zephyr/build.sh
index 9d82068a9..0cb8a825a 100755
--- a/samples/zephyr/build.sh
+++ b/samples/zephyr/build.sh
@@ -1,7 +1,10 @@
 #!/usr/bin/env bash
 
+export KONAN_DATA_DIR=$HOME/.konan
+export KONAN_DEPS=$KONAN_DATA_DIR/dependencies
```
so that Unix build works as well.

**Comment by olonho** [comment]
Tested that with the fix above we can build the final image and flash it into the card.

File: samples/zephyr/c_interop/platforms/stm32f4_disco.bat (hunk 1 above)
Comment (jwoolston) [reply samples/zephyr/c_interop/platforms/stm32f4_disco.bat]: Good call, copy paste error.

**Comment by jwoolston** [comment]
Requested changes complete.

**Pull request merged**